packages/palette/src/Theme.tsx

```diff
@@ -49,6 +49,12 @@ export const useTheme = <T extends TTheme>() => {
   return { theme }
 }
 
+/** Returns a config specific to the current theme */
+export const useThemeConfig = <T, U>({ v2, v3 }: { v2: T; v3: U }) => {
+  const { theme = { id: "v2" } } = useTheme()
+  return { v2, v3 }[theme.id as "v2" | "v3"]
```

Comment:
#nit Can we make this form a little more conventional

Reply:
WHAT HAPPENS WHEN WE HAVE THREE OR FOUR THEMES?

(I kid, will make it a ternary)